lib/rmt/downloader.rb

```diff
@@ -8,34 +8,39 @@
 require 'rmt/deduplicator'
 
 class RMT::Downloader
-  class Exception < RuntimeError
-    attr_reader :http_code, :response
-
-    def initialize(message, response: nil)
-      @response = response
-      @http_code = response&.code
-      super(message)
-    end
-  end
+  RETRIES = 4
+  RETRY_DELAY_SECONDS = 1
 
   attr_accessor :concurrency, :logger, :auth_token
 
   def initialize(logger:, auth_token: nil, track_files: true)
     Typhoeus::Config.user_agent = "RMT/#{RMT::VERSION}"
+    Typhoeus::Config.verbose = Settings.try(:http_client).try(:verbose)
+
     @concurrency = 4
     @auth_token = auth_token
     @logger = logger
     @track_files = track_files
     @queue = []
   end
 
-  def download(file_reference)
+  def download(file_reference, retries: RETRIES)
```

Comment:
It's called recursively now, that's why I send the retry count in the method.

Reply:
I understand, it can be done like that when using `retry`, but please check the current code in the pr: Because the request retries are triggered in the Fiber now, it's re-enqueuing a new request and needs to hand over the decremented retry count.

Reply:
Got it 

Could lines 81-82 of `downloader` 
```
  request = create_fiber_request(file_reference, failed_downloads: failed_downloads, retries: (retries - 1))         
  @hydra.queue(request) if request
  ```
  be changed in favour of `process_queue` (adapting it accorddingy) ?

Reply:
This doesn't work here, because I think hydra will not run another request in the same context.